lib/screens/auth/mobile_auth_screen.dart

```diff
@@ -0,0 +1,296 @@
+          children: [
+            Image.asset(
+              'assets/images/calendar_icon.png',
+              width: AppDimensions.imageSize40,
+              height: AppDimensions.imageSize40,
+            ),
+            const SizedBox(width: 10),
+            Text(
+              'general.label_uppercase'.tr(),
+              style: AppTextStyle.rubicRegularMobileLabel,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _inputBar(
+    Icon icon,
+    String hintText,
+    TextEditingController textEditingController,
+    bool obscure,
+  ) {
+    return Container(
+      padding: const EdgeInsets.symmetric(
+        horizontal: AppDimensions.padding20,
+      ),
+      child: TextField(
+        controller: textEditingController,
+        obscureText: obscure,
+        style: AppTextStyle.rubicRegular20,
+        decoration: InputDecoration(
+          hintStyle: AppTextStyle.rubicRegularHint20,
+          hintText: hintText,
+          focusedBorder: const OutlineInputBorder(
+            borderSide: BorderSide(
+              color: AppColors.white,
+              width: 3,
+            ),
+          ),
+          enabledBorder: const OutlineInputBorder(
+            borderSide: BorderSide(
+              color: AppColors.white54,
+              width: 1,
+            ),
+          ),
+          prefixIcon: Padding(
+            padding: const EdgeInsets.symmetric(
+              horizontal: AppDimensions.padding6,
+            ),
+            child: IconTheme(
+              child: icon,
+              data: const IconThemeData(color: AppColors.white),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _authForm({
+    required String label,
+    required BuildContext context,
+    required VoidCallback func,
+  }) {
+    return Column(
+      children: <Widget>[
+        Padding(
+          padding: const EdgeInsets.only(
+            top: AppDimensions.padding10,
+            bottom: AppDimensions.padding20,
+          ),
+          child: _inputBar(
+            const Icon(Icons.email),
+            'general.email_uppercase'.tr(),
+            _emailController,
+            false,
+          ),
+        ),
+        Padding(
+          padding: const EdgeInsets.only(
+            bottom: AppDimensions.padding20,
+          ),
+          child: _inputBar(
+            const Icon(Icons.lock),
+            'general.password_uppercase'.tr(),
+            _passwordController,
+            true,
+          ),
+        ),
+        const SizedBox(height: AppDimensions.padding20),
+        Container(
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(AppDimensions.borderRadius4),
+            color: AppColors.white,
+          ),
+          height: 50,
+          width: MediaQuery.of(context).size.width - 40,
+          child: _loginButton(label, func),
+        ),
+      ],
+    );
+  }
+
+  Widget _loginButton(String textLabel, VoidCallback onTap) {
```

Comment:
This widget is duplicated on several pages, it needs to be moved to a separate file. 
Also, the parameters must be named and required